src/rebar_app_discover.erl

```diff
@@ -411,12 +411,13 @@ try_handle_app_src_file(_AppInfo, _, _AppDir, _AppSrcFile, valid) ->
     false;
 try_handle_app_src_file(AppInfo, _, AppDir, [File], Validate) when Validate =:= invalid
                                                                  ; Validate =:= all ->
-    AppInfo1 = create_app_info(AppInfo, AppDir, File),
+    AppInfo1 = rebar_app_info:app_file(AppInfo, undefined),
```

Comment:
This fix with the follow-up validate app_info in the 'create_app_info' call, then this is equivalent to always returning ` false` on the validation. I think this is the wrong fix.

Reply:
Nevermind, saw the reloading that happens deeper down. Odd but I can see why it works; forces a re-parse. Alright, will tie some ends together and merge this stuff.